doc/source/raysgd/raysgd_pytorch.rst

```diff
@@ -3,114 +3,343 @@ RaySGD Pytorch
 
 .. warning:: This is still an experimental API and is subject to change in the near future.
 
-.. tip:: Help us make RaySGD better; take this 1 minute `User Survey <https://forms.gle/26EMwdahdgm7Lscy9>`_!
+.. tip:: Get in touch with us if you're using/considering using `RaySGD <https://forms.gle/26EMwdahdgm7Lscy9>`_!
 
-Ray's ``PyTorchTrainer`` simplifies distributed model training for PyTorch. The ``PyTorchTrainer`` is a wrapper around ``torch.distributed.launch`` with a Python API to easily incorporate distributed training into a larger Python application, as opposed to needing to execute training outside of Python.
+.. image:: raysgd-pytorch.svg
+    :align: center
 
-----------
+The RaySGD ``PyTorchTrainer`` simplifies distributed model training for PyTorch. The ``PyTorchTrainer`` is a wrapper around ``torch.distributed.launch`` with a Python API to easily incorporate distributed training into a larger Python application, as opposed to needing to execute training outside of Python.
 
-**With Ray**:
+Under the hood, ``PytorchTrainer`` will create *replicas* of your model (controlled by ``num_replicas``) which are each managed by a Ray actor.
 
-Wrap your training with this:
+.. image:: raysgd-actors.svg
+    :align: center
+
+Setting up training
+-------------------
+
+The ``PyTorchTrainer`` can be constructed with functions that construct components of the training script. Specifically, it needs constructors for the Model, Data, Optimizer, and Loss. This is so that we can create replicated copies of the neural network/optimizer across different devices and machines.
```

Comment:
In this paragraph, we should also mention the cases where we take in different types for each argument.